monitoring/monitoring_on.go

```diff
@@ -0,0 +1,46 @@
+// +build monitoring
+
+package monitoring
+
+import (
+	// "fmt"
+	"net/http"
+	"sync"
+
+	"google.golang.org/grpc"
+
+	"github.com/grpc-ecosystem/go-grpc-prometheus"
+	"github.com/lightningnetwork/lnd/lncfg"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+// Enabled signifies whether the monitoring tag was specified when building lnd
+// and whether to automatically export Prometheus metrics.
+const Enabled = true
+
+var started sync.Once
+
+// GetPromInterceptors returns the set of interceptors for Prometheus
+// monitoring.
+func GetPromInterceptors() ([]grpc.UnaryServerInterceptor, []grpc.StreamServerInterceptor) {
+	unaryInterceptors := []grpc.UnaryServerInterceptor{
+		grpc_prometheus.UnaryServerInterceptor,
+	}
+	streamInterceptors := []grpc.StreamServerInterceptor{
+		grpc_prometheus.StreamServerInterceptor,
+	}
+	return unaryInterceptors, streamInterceptors
+}
+
+// ExportPrometheusMetrics sets server options, registers gRPC metrics and
+// launches the Prometheus exporter on the specified address.
+func ExportPrometheusMetrics(grpcServer *grpc.Server, cfg lncfg.Prometheus) {
+	started.Do(func() {
+		grpc_prometheus.Register(grpcServer)
+
```

Comment:
Should adds some logging here to users can be sure that things have actually activated.